src/ynEIGEN/withdrawalsProcessor.sol

```diff
@@ -0,0 +1,364 @@
+    function getQueueWithdrawalsArgs() view returns (IERC20 _asset, uint256 _minNodeShares, uint256 _maxSharesToWithdraw) {
+
+        // get `_asset`
+        {
+            IERC20[] memory _assets = assetRegistry.getAssets();
+            uint256[] memory _balances = yneigen.assetBalances(_assets);
+
+            uint256 _highestBalance;
+            uint256 _assetsLength = _assets.length;
+            for (uint256 i = 0; i < _assetsLength; ++i) {
+                if (_balances[i] > _highestBalance) {
+                    _highestBalance = _balances[i];
+                    _asset = _assets[i];
+                }
+            }
+        }
+
+        ITokenStakingNode[] memory _nodes = tokenStakingNodesManager.getAllNodes();
+        uint256 _nodesLength = _nodes.length;
+        uint256[] memory _nodesShares = new uint256[](_nodesLength);
+
+        // get `_minNodeShares`
+        {
+            IStrategy _strategy = ynStrategyManager.strategies(_asset);
+
+            uint256 _highest;
+            uint256 _secondHighest;
+            for (uint256 i = 0; i < _nodesLength; ++i) {
+                uint256 _nodeShares = wrapper.toUserAssetAmount(_asset, _strategy.userUnderlyingView(address(_nodes[i]))); // @todo
+                if (_nodeShares > _highest) {
+                    _secondHighest = _highest;
+                    _highest = _nodeShares;
+                } else if (_nodeShares > _secondHighest && _nodeShares < _highest) {
+                    _secondHighest = _nodeShares;
+                }
+
+                _nodesShares[i] = _nodeShares;
+            }
+
+            _minNodeShares = _secondHighest;
+        }
+
+        // get `_maxSharesToWithdraw`
+        {
+            for (uint256 i = 0; i < _nodesLength; ++i) {
+                if (nodeShares[i] > _minNodeShares) {
+                    uint256 _availableToWithdraw = nodeShares[i] - _minNodeShares;
+                    if (_availableToWithdraw < _maxSharesToWithdraw || _maxSharesToWithdraw == 0) {
+                        _maxSharesToWithdraw = _availableToWithdraw;
+                    }
+                }
+            }
+        }
+    }
+
+    //
+    // Processor functions - mutative
+    //
+
+    /// @notice Queues withdrawals
+    /// @dev Reverts if the total pending withdrawal requests are below the minimum threshold
+    /// @dev Saves the queued withdrawals together in a batch, to be completed in the next step (`completeQueuedWithdrawals`)
+    /// @dev Before calling this function, call `getQueueWithdrawalsArgs()` to get the arguments
+    /// @param _asset The asset to withdraw
+    /// @param _minNodeShares The minimum amount of shares a node must have so that its shares can be withdrawn
+    /// @param _maxSharesToWithdraw The maximum amount of shares that can be withdrawn from a node
+    /// @return True if all pending withdrawal requests were queued, false otherwise
+    function queueWithdrawals(
+        IERC20 _asset,
+        uint256 _minNodeShares,
```

Comment:
does it make more sense if this input is arrays?

As in one item per node for more control